.github/workflows/build.yml

```diff
@@ -40,11 +39,48 @@ jobs:
       - name: docker login
         uses: aws-actions/amazon-ecr-login@v1
 
+      # for staging
+      # gather metadata from git & github actions to reference in docker
+      - name: git & github metadata staging
+        id: metadata_staging
+        uses: docker/metadata-action@v3
+        if: ${{ github.event.inputs.is_production == false }}
+        with:
+          images: ${{ env.AWS_ACCOUNT_ID }}.dkr.ecr.${{ env.AWS_REGION }}.amazonaws.com/openverse/frontend
+          tags: |
+            latest=auto
+            type=edge,branch=main
+
       # build a docker image
       - name: build docker image
         uses: docker/build-push-action@v2
+        if: ${{ github.event.inputs.is_production == false }}
         with:
           context: .
-          tags: ${{ steps.metadata.outputs.tags }}
-          labels: ${{ steps.metadata.outputs.labels }}
+          tags: ${{ steps.metadata_staging.outputs.tags }}
+          labels: ${{ steps.metadata_staging.outputs.labels }}
           push: true
+
+      # for production
+      # gather metadata from git & github actions to reference in docker
+      - name: git & github metadata production
+        id: metadata_production
+        uses: docker/metadata-action@v3
+        if: ${{ github.event.inputs.is_production == true }}
+        with:
+          images: ${{ env.AWS_ACCOUNT_ID }}.dkr.ecr.${{ env.AWS_REGION }}.amazonaws.com/openverse/frontend
+          tags: |
+            latest=false
+            type=raw,value=stable
+            type=semver,pattern={{version}}
+
+      # build a docker image
+      - name: build docker image
+        uses: docker/build-push-action@v2
+        if: ${{ github.event.inputs.is_production == true }}
+        with:
+          context: .
+          tags: ${{ steps.metadata_production.outputs.tags }}
+          labels: ${{ steps.metadata_production.outputs.labels }}
+          push: true
```

Comment:
Does this push trigger a deployment of our AWS infrastructure? Is that encapsulated in this workflow or will that be future work?

Reply:
The deployment is made by CodePipeline in our AWS. I just push the image to ECR and the event will automatically be triggered internally.

That's why I opened a [discussion](https://github.com/WordPress/openverse/discussions/135) a few days ago, because I would need to change this part and build some communication between CodePipeline and ghcr.io events.